### ISSUE TITLE
MA0015 false positive for record properties

### ISSUE THREAD

**Issue by robertlyson** [meziantou/Meziantou.Analyzer#558]
### Version of the Meziantou.Analyzer NuGet package

2.0.62

### Rule Identifier

MA0015

### Target Framework

net6

### C# Language version

11

### Description

Hello 👋 ,

MA0015 gives me false positive when using `nameof` in below context. I may be wrong, but following usage of `nameof` should not be marked as violation of MA0015 rule. 

Thanks for creating this package with wonderful analyzers!

### Reproduction Steps

Minimal code:

```c#
internal sealed record ManuscriptId(int Id)
{
    public int Id { get; } = Id > 0
        ? Id
        : throw new ArgumentOutOfRangeException(
            paramName: nameof(Id), Id, message: "Must be greater than 0");
}
```

**Comment by meziantou** [comment]
The fix will be included in 2.0.66. It should also support primary constructor (C# 12)

**Comment by robertlyson** [comment]
Thank you @meziantou for addressing the issue so quickly!
